frontend/src/components/MeetingRoom/DeviceSettingsMenu/DeviceSettingsMenu.tsx

```diff
@@ -0,0 +1,118 @@
+
+export type DeviceSettingsMenuProps = {
+  isAudioControl?: boolean;
+  handleToggle: () => void;
+  isOpen: boolean;
+  anchorRef: RefObject<HTMLInputElement>;
+  handleClose: (event: MouseEvent | TouchEvent) => void;
+  setIsOpen: Dispatch<SetStateAction<boolean>>;
+};
+
+const DeviceSettingsMenu = ({
+  isAudioControl,
+  handleToggle,
+  isOpen,
+  anchorRef,
+  handleClose,
+  setIsOpen,
+}: DeviceSettingsMenuProps): ReactElement | false => {
+  const theme = useTheme();
+  const customLightBlueColor = 'rgb(138, 180, 248)';
+
+  useDropdownResizeObserver({ setIsOpen, dropDownRefElement: anchorRef.current });
+
+  const renderSettingsMenu = () => {
+    if (isAudioControl) {
+      return (
+        <>
+          <InputDevices handleToggle={handleToggle} customLightBlueColor={customLightBlueColor} />
+          <OutputDevices handleToggle={handleToggle} customLightBlueColor={customLightBlueColor} />
+          <ReduceNoiseTestSpeakers customLightBlueColor={customLightBlueColor} />
+        </>
+      );
+    }
+
+    return (
+      <>
+        <VideoDevices handleToggle={handleToggle} customLightBlueColor={customLightBlueColor} />
+        {hasMediaProcessorSupport() && (
+          <>
+            <DropdownSeparator />
+            <VideoDevicesOptions customLightBlueColor={customLightBlueColor} />
+          </>
+        )}
+      </>
+    );
+  };
+
+  return (
+    <Popper
+      data-testid={
+        isAudioControl ? 'audio-settings-devices-dropdown' : 'video-settings-devices-dropdown'
+      }
+      open={isOpen}
+      anchorEl={anchorRef.current}
+      transition
+      disablePortal
+      placement="bottom-start"
+    >
+      {({ TransitionProps, placement }: PopperChildrenProps) => (
+        <Grow
+          {...TransitionProps}
+          style={{
+            transformOrigin: placement === 'bottom' ? 'center top' : 'center bottom',
+          }}
+        >
+          <div className="text-left font-normal">
+            <ClickAwayListener onClickAway={handleClose}>
+              <Paper
+                sx={{
+                  backgroundColor: 'rgb(32, 33, 36)',
+                  color: '#fff',
+                  padding: { xs: 1, sm: 2 }, // responsive padding
+                  borderRadius: 2,
+                  zIndex: 1,
+                  transform: isAudioControl
+                    ? 'translateY(-2%) translateX(5%)'
+                    : 'translateY(-5%) translateX(-15%)', // default transform
+                  [theme.breakpoints.down(741)]: {
+                    transform: isAudioControl
+                      ? 'translateY(-2%) translateX(-10%)'
+                      : 'translateY(-5%) translateX(-40%)',
+                  },
+                  [theme.breakpoints.down(450)]: {
+                    transform: isAudioControl
+                      ? 'translateY(-2%) translateX(-5%)'
+                      : 'translateY(-5%) translateX(-5%)',
+                  },
```

Comment:
Is it possible to explain some of these values or where we arrived at each breakpoint? I'm guessing, no.

Reply:
just trial and error, really